src/pages/AddItem.js

```diff
@@ -2,22 +2,72 @@ import React, { useState, useEffect } from 'react';
 import { withFirestore } from 'react-firestore';
 import AddItemHeader from './AddItemHeader';
 import getToken from '../lib/token';
+import './AddItem.css';
+import ItemError from './ItemError';
 
 const expectedPurchase = { soon: 7, kindOfSoon: 14, notSoon: 30 };
 
 const AddItem = ({ firestore }) => {
   const [name, setName] = useState('');
-
   const [nextExpectedPurchase, setNextExpectedPurchase] = useState(0);
+  const [duplicate, setDuplicate] = useState(false);
 
   const initialToken = () => window.localStorage.getItem('token') || getToken();
   const [token] = useState(initialToken);
   useEffect(() => {
     window.localStorage.setItem('token', token);
   }, [token]);
 
-  const addItem = name => {
-    firestore.collection('items').add({ name, token, nextExpectedPurchase });
+  // const addItem = name => {
+  //   firestore.collection('items').add({ name, token, nextExpectedPurchase });
+  // };
+
+  //normalized function
+  const normalizeName = name => {
+    name = name.toLowerCase().trim();
+    let normalizedName = '';
+    let symbol = `.,;:!?"`;
+    for (let i = 0; i < name.length; i++) {
+      if (!symbol.includes(name[i])) {
+        normalizedName += name.slice(i, i + 1);
+      }
+    }
+    return normalizedName;
+  };
+
+  const addItem = (normalizedName, nextExpectedPurchase) => {
+    firestore
+      .collection('items')
+      .doc(name)
+      .set({ name: '' });
+
+    //set doc ID = to item name
+    //Nikema: I'm grabbing the database wrong and I don't know what it is could you take a look?
```

Comment:
Check out this document in regards to taking a shapshot of the database [firebase snapshot](https://firebase.google.com/docs/firestore/query-data/listen)

I think the problem is i'm confusing myself on the naming of the data structure we are using. Maybe with your set of fresh eyes you can spot it.